src/test/java/seedu/address/logic/commands/UntagCommandTest.java

```diff
@@ -0,0 +1,196 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_FRIEND;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HUSBAND;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_STUDENT;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_TEACHING_ASSISTANT;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_WIFE;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_PERSON;
+import static seedu.address.testutil.TypicalPersons.getTaggedTypicalAddressBook;
+
+import java.util.HashSet;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+import seedu.address.testutil.EditPersonDescriptorBuilder;
+import seedu.address.testutil.PersonBuilder;
+
+class UntagCommandTest {
+
+    private Model generateFriendTaggedModel() {
+        return new ModelManager(getTaggedTypicalAddressBook(VALID_TAG_FRIEND), new UserPrefs());
+    }
+
+    private Model generateFriendAndHusbandTaggedModel() {
+        return new ModelManager(getTaggedTypicalAddressBook(VALID_TAG_FRIEND, VALID_TAG_HUSBAND), new UserPrefs());
+    }
+
+    private Model generateFriendHusbandTeachingAssistantStudentTaggedModel() {
+        return new ModelManager(getTaggedTypicalAddressBook(VALID_TAG_FRIEND, VALID_TAG_HUSBAND,
+                VALID_TAG_TEACHING_ASSISTANT, VALID_TAG_STUDENT), new UserPrefs());
+    }
+
+    @Test
+    void execute_singleTagNoRemainingTagsUnfilteredList_success() {
+        Model model = generateFriendTaggedModel();
+        Person firstPerson = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+
+        PersonBuilder personInList = new PersonBuilder(firstPerson);
+        Person editedPerson = personInList.withTags().build();
+
+        UntagCommand.EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder()
+                .withTags(VALID_TAG_FRIEND).build();
+        UntagCommand untagCommand = new UntagCommand(INDEX_FIRST_PERSON, descriptor);
+
+        String expectedMessage = String.format(UntagCommand.MESSAGE_REMOVE_PERSON_SUCCESS,
+                editedPerson.getName(), UntagCommand.getRemovedTags(descriptor));
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
```

Comment:
Should we be building the expected model based on a copy of the test model
Consider building from ```generateFriendTaggedModel();``` instead

Reply:
I mean this was what the editCommandTest did also